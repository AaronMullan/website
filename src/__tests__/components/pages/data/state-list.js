import React from 'react'
import renderer from 'react-test-renderer'
import StateList from '../../../../components/pages/data/state-list'
import sampleState from 'sample-state'

describe('Components : Pages : Data : State data', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <StateList
          states={[{ node: sampleState.state }]}
          stateData={[{ node: sampleState.stateData }]}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
