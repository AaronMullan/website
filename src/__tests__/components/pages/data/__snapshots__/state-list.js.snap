// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Data : State data renders correctly 1`] = `
<div
  className="item"
  id="state-list-AL"
>
  <div
    className="state-header header"
  >
    <h3
      id="state-al"
    >
      <a
        href="/data/state/alabama"
      >
        Alabama
      </a>
    </h3>
    <p
      className="state-grade stateGrade"
    >
      <span>
        Current 
        <a
          href="/state-grades"
        >
          data quality grade
        </a>
        :
      </span>
      <img
        alt="Grade na"
        src="test-file-stub"
        width="28px"
      />
    </p>
  </div>
  <div>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            className="alignLeft borderLeft"
            scope="col"
          >
            <span
              role="text"
            >
              <span
                className="headerLabel headerLabel1"
              >
                Cases
              </span>
            </span>
          </th>
          <th
            className="alignLeft borderLeft"
            scope="col"
          >
            <span
              role="text"
            >
              Negative
              <span
                className="headerLabel headerLabel2"
              >
                Tests
              </span>
            </span>
          </th>
          <th
            className=""
            scope="col"
          >
            <span
              role="text"
            >
              Pending
              <span
                className="headerLabelHidden headerLabel2"
              >
                Tests
              </span>
            </span>
          </th>
          <th
            className="alignLeft borderLeft"
            scope="col"
          >
            <span
              role="text"
            >
              Currently
              <span
                className="headerLabel headerLabel2"
              >
                Hospitalized
              </span>
            </span>
          </th>
          <th
            className=""
            scope="col"
          >
            <span
              role="text"
            >
              Cumulative
              <span
                className="headerLabelHidden headerLabel2"
              >
                Hospitalized
              </span>
            </span>
          </th>
          <th
            className="alignLeft borderLeft"
            scope="col"
          >
            <span
              role="text"
            >
              Currently
              <span
                className="headerLabel headerLabel2"
              >
                In ICU
              </span>
            </span>
          </th>
          <th
            className=""
            scope="col"
          >
            <span
              role="text"
            >
              Cumulative
              <span
                className="headerLabelHidden headerLabel2"
              >
                In ICU
              </span>
            </span>
          </th>
          <th
            className="alignLeft borderLeft"
            scope="col"
          >
            <span
              role="text"
            >
              Currently
              <span
                className="headerLabel headerLabel2"
              >
                On Ventilator
              </span>
            </span>
          </th>
          <th
            className=""
            scope="col"
          >
            <span
              role="text"
            >
              Cumulative
              <span
                className="headerLabelHidden headerLabel2"
              >
                On Ventilator
              </span>
            </span>
          </th>
          <th
            className="alignLeft borderLeft"
            scope="col"
          >
            <span
              role="text"
            >
              Recovered
              <span
                className="headerLabel headerLabel2"
              >
                Outcomes
              </span>
            </span>
          </th>
          <th
            className=""
            scope="col"
          >
            <span
              role="text"
            >
              Deaths
              <span
                className="headerLabelHidden headerLabel2"
              >
                Outcomes
              </span>
            </span>
          </th>
          <th
            className="alignLeft borderLeft wide"
            scope="col"
          >
            <span
              role="text"
            >
              Positive + Negative
              <span
                className="headerLabel headerLabel1"
              >
                Total Test Results
              </span>
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="borderLeft"
          >
            4,572
          </td>
          <td
            className="borderLeft"
          >
            33,318
          </td>
          <td
            className=""
          >
            N/A
          </td>
          <td
            className="borderLeft"
          >
            N/A
          </td>
          <td
            className=""
          >
            594
          </td>
          <td
            className="borderLeft"
          >
            N/A
          </td>
          <td
            className=""
          >
            247
          </td>
          <td
            className="borderLeft"
          >
            N/A
          </td>
          <td
            className=""
          >
            148
          </td>
          <td
            className="borderLeft"
          >
            N/A
          </td>
          <td
            className=""
          >
            151
          </td>
          <td
            className="borderLeft"
          >
            37,890
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="detail-text detailText"
    >
      Last updated: 
      Fri Apr 17 2020 12:00 am
       
      <abbr
        aria-label="Eastern Time"
        title="Eastern Time"
      >
        ET
      </abbr>
    </div>
  </div>
  <ul
    className="list"
  >
    <li>
      <a
        href="https://twitter.com/@alpublichealth"
        onClick={[Function]}
      >
        <span
          className="a11y-only"
        >
          Alabama
          's 
        </span>
        Official Twitter
      </a>
      <span>
        •
      </span>
    </li>
    <li>
      <a
        href="https://alpublichealth.maps.arcgis.com/apps/opsdashboard/index.html#/6d2771faa9da4a2786a509d82c8cf0f7"
        onClick={[Function]}
      >
        <span
          className="a11y-only"
        >
          Alabama
          's 
        </span>
        Best Current Data Source
      </a>
      <span>
        •
      </span>
    </li>
    <li>
      <a
        href="/data/state/alabama#historical"
      >
        Historical Data
        <span
          className="a11y-only"
        >
          for 
          Alabama
        </span>
      </a>
    </li>
  </ul>
  <div
    className="container container"
  >
    <div
      className="narrow"
    >
      <div
        className="content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Negatives = (Totals - Positives) 
Positives occasionally update before totals do; do not revise negatives down, keep the last calculated negative.
Last Updated (Local Time) = (current date + time = 00:00)</p>
",
          }
        }
      />
    </div>
  </div>
  <a
    className="state-top-link topLink"
    href="#reach-skip-nav"
    title="top"
  >
    Back to top.
  </a>
</div>
`;
